### DISCUSSION TITLE
How to stop adding a duplicate-letter to the filename with addto?

### DISCUSSION THREAD

**Discussion by tuurep** [papis/papis#762] [Q&A] [answered]
Hello,

I'm trying to add a pdf to a document

Ideally I'd want the directory to be named `[author]-[year]` and the associated pdf as `[author]-[year].pdf`

I was very close to achieving this with the config options:

```
add-folder-name = {doc[author_list][0][family]}-{doc[year]}
add-file-name = {doc[author_list][0][family]}-{doc[year]}
```

However, no matter what I do, the file added with addto will append `-a` to the document (as in `[author]-[year]-a.pdf` or `foobar-a.pdf`)

I can see the use when I would add more than one pdf to the same document, but is it possible to not add it if it's the first pdf to be added to a document?

**Comment by alexfikl** [comment, marked as answer]
That was a bug in how the indexing happened..

There's a fix in #764 that seems to work. Let me know if you get a chance to test it out!

**Reply by tuurep** [reply]
Oh.. I legitimately thought I was doing something wrong :)

Thank you! I'll test it after I figure out how to build outside of package managers. I'll let you know.

**Reply by alexfikl** [reply]
Sure thing! I'll go ahead and merge that so that it's in the next release anyway. :grin:

**Reply by tuurep** [reply]
Ok I was able to test this today, yeah works great!

Additional thoughts:

It would match my brain best if in the case of adding pdf number 2, the original would be *renamed* to `...-a.pdf` and second one would be `...-b.pdf`

And if I'd add another filetype, such as markdown, they could have the same name but with different extension:

```
[author]-[year]/
    [author]-[year].pdf
    [author]-[year].md
```

But you've fixed my immediate problem & quite a headache, so thanks so much :smile:

**Reply by tuurep** [reply]
Hmm.. what do you think about this case though? My comments above are more of a feature request/idea, but

when you use `papis addto -f foo.txt --file-name some-specific-thing.txt`

Can't avoid adding the duplicate letter even though you'd use the `--file-name` specifically to have a different name.

**Reply by alexfikl** [reply]
Hm, maybe a simple naming scheme would be to add a suffix only if the file already exists? That should handle the PDF and markdown file case and should also leave user-provided file names alone.

**Reply by tuurep** [reply]
Sounds like a great idea to me

**Reply by alexfikl** [reply]
FYI, #831 should hopefully address this (once it gets in).

**Reply by tuurep** [reply]
Thanks! I saw even before your comment :) been following the recent changes in the repo and it's all looking great.
